packages/linter.rb

```diff
@@ -1,91 +0,0 @@
-require 'package'
-
-class Linter < Package
-  description 'Comprehensive linter and code analysis for various file types.'
-  homepage 'https://github.com/chromebrew/chromebrew'
-  version '1.4'
-  license 'GPL-3'
-  compatibility 'all'
-  source_url 'SKIP'
-
-  no_compile_needed
-
-  depends_on 'py3_codespell'
-  depends_on 'ruby_mdl'
-  depends_on 'ruby_rubocop'
-  depends_on 'ruby_yaml_lint'
-  depends_on 'shellcheck'
-
-  def self.build
-    linter = <<~EOF
-      #!/bin/bash
-      export PATH=#{CREW_PREFIX}/bin:$PATH
-      if test "$1"; then
-        files="$*"
-      else
-        files="$(git ls-files -om | xargs)"
-      fi
-      for file in $files; do
-        ext="${file##*.}"
-        if ! [[ "$ext" =~ ^(md|rb|sh|yml|yaml)$ ]]; then
-          type="$(file -b "$file" | cut -d' ' -f1)"
-          case $type in
-            Bourne-Again)
-              ext="sh"
-              ;;
-            HTML)
-              ext="md"
-              ;;
-            Ruby)
-              ext="rb"
-              ;;
-          esac
-        fi
-        case $ext in
-          md)
-            mdl -c "$HOME/.mdlrc" "$file"
-            ;;
-          rb)
-            rubocop "$file"
-            ;;
-          sh)
-            shellcheck "$file"
-            ;;
-          yml|yaml)
-            yaml-lint "$file"
-            ;;
-          *)
-            echo "Unable to check syntax of $file."
-            ;;
-        esac
-        codespell "$file"
```

Comment:
This goes away too with this script.

Reply:
We haven't used codespell since 2022... #7542

Reply:
Well, we should start using it again.  I've seen a lot of poorly spelled comments.  Where do you get these ideas about ditching stuff?  FYI, this is the linter script, not to be run in Github Actions but rather, locally.

Reply:
You're welcome to write the codespell CI and fix the issues that led to you removing it last time...

> FYI, this is the linter script, not to be run in Github Actions but rather, locally.

What is the actual usecase for this script? If its for developer convenience, the pre-commit hooks replace that.

If its for users, it probably shouldn't be-- why would we ship a package which runs four random linters, one of them with an opinionated configuration, in the current (git) directory?

Reply:
The pre-commit hooks are not convenient.  For one, the package is not installed with buildessential and even if it was, there is no explanation of the need to run `pre-commit install` to make it useful other than a comment that nobody will read unless they look in the package.  That should be at the very least a postinstall message.

Reply:
There is nothing random or opinionated about these linters.  You are just like my wife who finds it easy to throw away things that belong to others. :scream_cat:

Reply:
The `linter` package shouldn't be a package at all-- this is not a program available anywhere else, this is an out-of-date convenience script. If you find this convenient, keep it on your local machine.

Reply:
You mean just like you added 2 more opinionated exclusions to the GitHub Actions rules?  It doesn't hurt to keep this package.

Reply:
> It doesn't hurt to keep this package.

Again, this is an outdated convenience script. This should not be a package.